Doc/library/datetime.rst

```diff
@@ -848,19 +972,38 @@ Other constructors, all class methods:
 
 .. classmethod:: datetime.fromisoformat(date_string)
 
-  Return a :class:`datetime` corresponding to a *date_string* in one of the
-  formats emitted by :meth:`date.isoformat` and :meth:`datetime.isoformat`.
-  Specifically, this function supports strings in the format(s)
-  ``YYYY-MM-DD[*HH[:MM[:SS[.fff[fff]]]][+HH:MM[:SS[.ffffff]]]]``,
-  where ``*`` can match any single character.
+   Return a :class:`datetime` corresponding to a *date_string* in one of the
+   formats emitted by :meth:`date.isoformat` and :meth:`datetime.isoformat`.
+
+   Specifically, this function supports strings in the format:
+
+   .. code-block:: none
+
+      YYYY-MM-DD[*HH[:MM[:SS[.fff[fff]]]][+HH:MM[:SS[.ffffff]]]]
 
-  .. caution::
+   where ``*`` can match any single character.
 
-    This does not support parsing arbitrary ISO 8601 strings - it is only intended
-    as the inverse operation of :meth:`datetime.isoformat`.
+   .. caution::
 
-  .. versionadded:: 3.7
+     This does *not* support parsing arbitrary ISO 8601 strings that do not meet
+     the exact format above. It is only intended as the inverse operation of
+     :meth:`datetime.isoformat`.
 
+   Examples::
+
+       >>> from datetime import datetime
+       >>> datetime.fromisoformat('2011-11-04')
+       datetime.datetime(2011, 11, 4, 0, 0)
+       >>> datetime.fromisoformat('2011-11-04T00:05:23')
+       datetime.datetime(2011, 11, 4, 0, 5, 23)
+       >>> datetime.fromisoformat('2011-11-04 00:05:23.283')
+       datetime.datetime(2011, 11, 4, 0, 5, 23, 283000)
+       >>> datetime.fromisoformat('2011-11-04 00:05:23.283+00:00')
+       datetime.datetime(2011, 11, 4, 0, 5, 23, 283000, tzinfo=datetime.timezone.utc)
+       >>> datetime.fromisoformat('2011-11-04T00:05:23.283+04:00')
+       datetime.datetime(2011, 11, 4, 0, 5, 23, 283000, tzinfo=datetime.timezone(datetime.timedelta(seconds=14400)))
```

Comment:
This causes the code block to start horizontally scrolling.

I'm not familiar enough with doctest syntax - is it valid to add a line break in one of the return values?

Reply:
Added a `doctest: +SKIP` in 09568f7ab1, but not sure how I feel about it.  Open to suggestions if you can think of an alternative or feel strongly it should just be removed.